recipes/samarium/all/conanfile.py

```diff
@@ -33,60 +35,48 @@ def _compilers_minimum_version(self):
             "apple-clang": "13",
         }
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def source(self):
+        get(self, **self.conan_data["sources"]
+            [str(self.version)], strip_root=True)
 
-    @property
-    def _build_subfolder(self):
-        return "build"
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
-    def configure(self):
-        if self.options.shared:
-            del self.options.fPIC
-
     def validate(self):
-        if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, self._min_cppstd)
+        if self.version == "1.0.0" and self.settings.os == "Macos":
+            raise ConanInvalidConfiguration("Macos not supported for v1.0.0")
 
         compiler = str(self.settings.compiler)
         if compiler not in self._compilers_minimum_version:
-            self.output.warn("Unknown compiler, assuming it supports at least C++20")
+            self.output.warn(
+                "Unknown compiler, assuming it supports at least C++20")
             return
 
-        version = tools.Version(self.settings.compiler.version)
+        version = Version(self.settings.compiler.version)
         if version < self._compilers_minimum_version[compiler]:
-            raise ConanInvalidConfiguration(f"{self.name} requires a compiler that supports at least C++20")
-
-    def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
```

Comment:
This was prefered

Reply:
@strangeQuark1041 please, revert it. When exporting using this way, we only export what we need, according to the package version.